frontend/src/context/OrderPayListProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import { PropsWithChildren, createContext, useState } from "react";
+
+import { OrderPayDetail, ProductDetail } from "../types";
+
+type InitialOrderPayList = {
+  getOrderPayList: () => OrderPayDetail[];
+  addProductToOrderPayList: (product: ProductDetail, quantity: number) => void;
+  updateProductInOrderPayList: (
+    product: ProductDetail,
+    quantity: number
+  ) => void;
+  deleteProductFromOrderPayList: (id: number) => void;
+  totalPrice: number;
+  updateTotalPrice: (price: number, quantity: number) => number;
+};
+
+const initialOrderPayList: InitialOrderPayList = {
+  getOrderPayList: () => [],
+  addProductToOrderPayList: () => null,
+  updateProductInOrderPayList: () => null,
+  deleteProductFromOrderPayList: () => null,
+  totalPrice: 0,
+  updateTotalPrice: () => 0,
+};
+
+export const OrderPayListContext = createContext(initialOrderPayList);
+
+export default function OrderPayListProvider({ children }: PropsWithChildren) {
+  const [orderPayList, setOrderPayList] = useState<OrderPayDetail[]>([]);
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  const getOrderPayList = (): OrderPayDetail[] => {
+    return orderPayList;
+  };
+
+  const addProductToOrderPayList = (
+    product: ProductDetail,
+    quantity: number
+  ) => {
+    setOrderPayList([{ ...product, quantity }, ...orderPayList]);
```

Comment:
prevState를 활용하는건 어떨까요!

```suggestion
    setOrderPayList((prevOrderList) => ([...prevOrderList, { ...product, quantity }]))
```